src/main/java/edu/cornell/kfs/concur/service/impl/ConcurAccountValidationServiceImpl.java

```diff
@@ -2,17 +2,173 @@
+        ValidationResult requiredFieldsValidationResult = checkRequiredAccountInfo(chartOfAccountsCode, accountNumber, objectCode);
+
+        if (requiredFieldsValidationResult.isNotValid()) {
+            return requiredFieldsValidationResult;
+        } else {
+            ValidationResult accountValidationResult = checkAccount(chartOfAccountsCode, accountNumber);
+            if (accountValidationResult.isNotValid()) {
+                return accountValidationResult;
+            } else {           
+                ValidationResult validationResult = new ValidationResult(true, new ArrayList<String>());
+                updateValidationResultAndAddErrorMessages(validationResult, checkObjectCode(chartOfAccountsCode, objectCode));
+                updateValidationResultAndAddErrorMessages(validationResult, checkSubAccount(chartOfAccountsCode, accountNumber, subAccountNumber));
+                updateValidationResultAndAddErrorMessages(validationResult, checkSubObjectCode(chartOfAccountsCode, accountNumber, objectCode, subObjectCode));
+                updateValidationResultAndAddErrorMessages(validationResult, checkProjectCode(projectCode));            
+                return validationResult;
+            }
+        }
+    }
+    
+    private void updateValidationResultAndAddErrorMessages(ValidationResult validationResult, ValidationResult specificCheckResult){
+        if (specificCheckResult.isNotValid()) {
+            validationResult.setValid(false);
+            validationResult.addMessages(specificCheckResult.getMessages());
+        }
+    }
+
+    public ValidationResult checkRequiredAccountInfo(String chartOfAccountsCode, String accountNumber, String objectCode) {
+        ValidationResult validationResult = new ValidationResult(true, new ArrayList<String>());
+        if (chartOfAccountsCode == null || chartOfAccountsCode.isEmpty()) {
+            validationResult.setValid(false);
+            validationResult.addMessage(errorMessageUtilsService.createErrorString(KFSKeyConstants.ERROR_REQUIRED, ConcurConstants.AccountingStringFieldNames.CHART));           
+        } 
+        if (accountNumber == null || accountNumber.isEmpty()) {
+            validationResult.setValid(false);
+            validationResult.addMessage(errorMessageUtilsService.createErrorString(KFSKeyConstants.ERROR_REQUIRED, ConcurConstants.AccountingStringFieldNames.ACCOUNT_NUMBER));
+        }
+        if (objectCode == null || objectCode.isEmpty()) {
+            validationResult.setValid(false);
+            validationResult.addMessage(errorMessageUtilsService.createErrorString(KFSKeyConstants.ERROR_REQUIRED, ConcurConstants.AccountingStringFieldNames.OBJECT_CODE));
+        }
 
         return validationResult;
     }
     
+    private ValidationResult checkMissingOrInactive(Inactivatable inactivatableObject, String missingMessage, String inactiveMessage){
```

Comment:
Not a big deal, but from a clean code perspective, I would probably move this method below where it is first called.